packages/assets-controllers/src/MultichainAssetsController/MultichainAssetsController.test.ts

```diff
@@ -0,0 +1,1539 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import type {
+  AccountAssetListUpdatedEventPayload,
+  CaipAssetTypeOrId,
+} from '@metamask/keyring-api';
+import { EthAccountType, EthMethod, EthScope } from '@metamask/keyring-api';
+import { KeyringTypes } from '@metamask/keyring-controller';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import type { PermissionConstraint } from '@metamask/permission-controller';
+import type { SubjectPermissions } from '@metamask/permission-controller';
+import type { Snap } from '@metamask/snaps-utils';
+import { useFakeTimers } from 'sinon';
+import { v4 as uuidv4 } from 'uuid';
+
+import type {
+  AssetMetadataResponse,
+  MultichainAssetsControllerMessenger,
+  MultichainAssetsControllerState,
+} from './MultichainAssetsController';
+import {
+  getDefaultMultichainAssetsControllerState,
+  MultichainAssetsController,
+} from './MultichainAssetsController';
+import { advanceTime } from '../../../../tests/helpers';
+import type {
+  ExtractAvailableAction,
+  ExtractAvailableEvent,
+} from '../../../base-controller/tests/helpers';
+
+const mockSolanaAccount = {
+  type: 'solana:data-account',
+  id: 'a3fc6831-d229-4cd1-87c1-13b1756213d4',
+  address: 'EBBYfhQzVzurZiweJ2keeBWpgGLs1cbWYcz28gjGgi5x',
+  options: {
+    scope: 'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp',
+  },
+  methods: ['sendAndConfirmTransaction'],
+  metadata: {
+    name: 'Snap Account 1',
+    importTime: 1737022568097,
+    keyring: {
+      type: 'Snap Keyring',
+    },
+    snap: {
+      id: 'local:http://localhost:8080',
+      name: 'Solana',
+      enabled: true,
+    },
+    lastSelected: 0,
+  },
+};
+
+const mockEthAccount = {
+  address: '0x807dE1cf8f39E83258904b2f7b473E5C506E4aC1',
+  id: uuidv4(),
+  metadata: {
+    name: 'Ethereum Account 1',
+    importTime: Date.now(),
+    keyring: {
+      type: KeyringTypes.snap,
+    },
+    snap: {
+      id: 'mock-eth-snap',
+      name: 'mock-eth-snap',
+      enabled: true,
+    },
+    lastSelected: 0,
+  },
+  scopes: [EthScope.Eoa],
+  options: {},
+  methods: [EthMethod.SignTypedDataV4, EthMethod.SignTransaction],
+  type: EthAccountType.Eoa,
+};
+
+const mockGetAssetsResult = [
```

Comment:
Nit: IMO we could name this differently just to keep the naming align with the actual controller's actions:

What about:

```suggestion
const mockHandleRequestOnAssetsLookupReturnValue = [
```